### PR TITLE
fix(secu): only return resources that you are allowed to see

### PR DESCRIPTION
**Pull request by adr-mo**
This PR intends to:
- Fix several security issues
- Cleaning dozens of coding style errors on files that were edited

For the security issue related to externalcmd an other PR will be made for the 19.10.x and 2.8.x will be made as in previous versions centcore was still used.

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by sc979** [comment]
Kindly rename this PR ;)

File: www/api/class/centreon_metric.class.php [hunk 1]
```diff
@@ -74,6 +74,12 @@ public function __construct()
      */
     public function getList()
     {
+        global $centreon;
+
+        $userId = $centreon->user->user_id;
+        $isAdmin = $centreon->user->admin;
```
Comment (sc979) [inline www/api/class/centreon_metric.class.php:80]: not really useful, as they're called only one time

File: www/api/class/centreon_metric.class.php [hunk 2]
```diff
@@ -125,8 +148,14 @@ protected function getListByService()
...
-            AND CONCAT(h.name," - ", s.description, " - ",  m.metric_name) LIKE :name
-            ORDER BY CONCAT(h.name," - ", s.description, " - ",  m.metric_name) COLLATE utf8_general_ci ';
+            AND CONCAT(h.name," - ", s.description, " - ",  m.metric_name) LIKE :name ';
+
+        if (!$isAdmin) {
+            $acl = new CentreonACL($userId, $isAdmin);
+            $aclMetrics .= 'AND s.service_id IN (' . $acl->getServicesString('ID', $this->pearDBMonitoring) . ') ';
+        }
+
+        $query .= ' ORDER BY CONCAT(h.name," - ", s.description, " - ",  m.metric_name) COLLATE utf8_general_ci ';
```
Comment (sc979) [inline www/api/class/centreon_metric.class.php:158]: could you try with the alias defined at line 144 ?
```suggestion
        $query .= ' ORDER BY fullname COLLATE utf8_general_ci ';
```

File: www/api/class/centreon_metric.class.php [hunk 3]
```diff
@@ -82,9 +88,20 @@ public function getList()
...
+        /**
+         * If ACLs on, then only return metrics linked to services that the user can see.
+         */
+        if (!$isAdmin) {
+            $acl = new CentreonACL($userId, $isAdmin);
+            $aclMetrics .= ' AND m.index_id = i.id AND i.service_id IN (' .
+                $acl->getServicesString('ID', $this->pearDBMonitoring) . ') ';
+        }
+
+        $query .= $aclMetrics;
```
Comment (sc979) [inline www/api/class/centreon_metric.class.php:103]:
```suggestion
            $query .= ' AND m.index_id = i.id AND i.service_id IN (' .
                $acl->getServicesString('ID', $this->pearDBMonitoring) . ') ';
        }

```

File: www/api/class/centreon_metric.class.php [hunk 4]
```diff
@@ -125,8 +148,14 @@ protected function getListByService()
             AND s.service_id = i.service_id
             AND h.enabled = 1
             AND s.enabled = 1
-            AND CONCAT(h.name," - ", s.description, " - ",  m.metric_name) LIKE :name
-            ORDER BY CONCAT(h.name," - ", s.description, " - ",  m.metric_name) COLLATE utf8_general_ci ';
+            AND CONCAT(h.name," - ", s.description, " - ",  m.metric_name) LIKE :name ';
+
+        if (!$isAdmin) {
+            $acl = new CentreonACL($userId, $isAdmin);
+            $aclMetrics .= 'AND s.service_id IN (' . $acl->getServicesString('ID', $this->pearDBMonitoring) . ') ';
```
Comment (sc979) [inline www/api/class/centreon_metric.class.php:155]:
```suggestion
            $query .= 'AND s.service_id IN (' . $acl->getServicesString('ID', $this->pearDBMonitoring) . ') ';
```

File: www/api/class/centreon_monitoring_poller.class.php [hunk 5]
```diff
@@ -59,6 +59,12 @@ public function __construct()
      */
     public function getList()
     {
+        global $centreon;
+
+        $userId = $centreon->user->user_id;
+        $isAdmin = $centreon->user->admin;
```
Comment (sc979) [inline www/api/class/centreon_monitoring_poller.class.php:65]: used only one time

File: www/api/class/centreon_metric.class.php [hunk 6]
```diff
@@ -110,6 +127,12 @@ public function getList()
      */
     protected function getListByService()
     {
+        global $centreon;
+
+        $userId = $centreon->user->user_id;
+        $isAdmin = $centreon->user->admin;
+        $aclMetrics = '';
```
Comment (sc979) [inline www/api/class/centreon_metric.class.php:134]:
```suggestion
```

File: www/api/class/centreon_monitoring_poller.class.php [hunk 7]
```diff
@@ -59,6 +59,12 @@ public function __construct()
...
+        $aclPollers = '';
```
Comment (sc979) [inline www/api/class/centreon_monitoring_poller.class.php:66]:
```suggestion
```

**Review by sc979 — CHANGES_REQUESTED** [review]

**Review by sc979 — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by loiclau — APPROVED** [review]

File: www/api/class/centreon_metric.class.php [hunk 8]
```diff
@@ -82,9 +84,19 @@ public function getList()
...
-            COLLATE utf8_bin as "metric_name" FROM `metrics`
-            WHERE metric_name LIKE :name
-            ORDER BY `metric_name` COLLATE utf8_general_ci ';
+            COLLATE utf8_bin as "metric_name", index_id FROM `metrics` as m, index_data i
+            WHERE metric_name LIKE :name ';
+
+        /**
+         * If ACLs on, then only return metrics linked to services that the user can see.
+         */
+        if (!$centreon->user->admin) {
```
Comment (callapa) [inline www/api/class/centreon_metric.class.php:93]: I think it would have been better to do a private method that you would have called

**Pull request merged**